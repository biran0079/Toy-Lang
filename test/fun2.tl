a=3;
fun f(){
  a=1;
  print(a);
  g();
  print(a);
}
fun g(){
  print(a);
  a=2;
  print(a);
}
f();
print(b);
