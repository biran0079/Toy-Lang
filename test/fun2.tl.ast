(stmts (fun (id f) (ids) (stmts (= (id a) (int 1)) (call (id print) (exps (id a))) (call (id g) (exps)) (tail_call (id print) (exps (id a))))) (fun (id g) (ids) (stmts (call (id print) (exps (id a))) (= (id a) (int 2)) (tail_call (id print) (exps (id a))))) (call (id f) (exps)))
