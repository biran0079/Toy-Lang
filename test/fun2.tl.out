1
none
2
1
